client/http_post.go

```diff
@@ -55,49 +56,47 @@ type PostContext struct {
 // retry here to eventually get through with the same client command ID and be
 // given the cached response.
 func HTTPPost(c PostContext, request, response gogoproto.Message, method fmt.Stringer) error {
-	retryOpts := c.RetryOpts
-	retryOpts.Tag = fmt.Sprintf("%s %s", c.Context.RequestScheme(), method)
-
 	// Marshal the args into a request body.
 	body, err := gogoproto.Marshal(request)
 	if err != nil {
 		return err
 	}
 
+	client, err := c.Context.GetHTTPClient()
+	if err != nil {
+		return err
+	}
+
 	url := c.Context.RequestScheme() + "://" + c.Server + c.Endpoint + method.String()
 
-	return retry.WithBackoff(retryOpts, func() (retry.Status, error) {
+	for r := retry.Start(c.RetryOpts); r.Next(); {
 		req, err := http.NewRequest("POST", url, bytes.NewReader(body))
```

Comment:
Due to the change in scope, I think this redefines `err` making the `err` inside the loop different from the `err` outside the loop. The subsequent `return err` at line 122 would always return `nil`. Am I missing something?